src/lwes_event.c

```diff
@@ -309,6 +329,30 @@ lwes_event_destroy
   return 0;
 }
 
+
+
+#define TYPED_EVENT_FIELD_TO_BYTES(typ, fail_ret) \
+  else if (tmp->type == LWES_TYPE_##typ)          \
+  {                                               \
+    if (!marshall_##typ                           \
+                (*((LWES_##typ *)tmp->value),     \
+                bytes, num_bytes, &tmpOffset))    \
+      { ret = fail_ret; }                         \
+  }                                               \
+  else if (tmp->type == LWES_TYPE_##typ##_ARRAY)  \
+  {                                               \
+    if (!marshall_array_attribute                 \
+          (tmp, bytes, num_bytes, &tmpOffset))    \
+      { ret = fail_ret-100; }                     \
+  }                                               \
+  else if (tmp->type == LWES_TYPE_N_##typ##_ARRAY)\
+  {                                               \
+    if (!marshall_array_attribute                 \
+          (tmp, bytes, num_bytes, &tmpOffset))    \
+      { ret = fail_ret-100; }                     \
```

Comment:
Should this be something other than 100 (which is the same as your clause right above this one)?  I'm not totally convinced there's any value for different errors, other than being able to test that explicit failures occur.  But since we already use different values for everything should this continue here?